src/backend/opengl/CMakeLists.txt

```diff
@@ -1,156 +1,83 @@
+        ${CMAKE_CURRENT_SOURCE_DIR}/colormap_impl.cpp
+        ${CMAKE_CURRENT_SOURCE_DIR}/common.hpp
+        ${CMAKE_CURRENT_SOURCE_DIR}/common.cpp
+        ${CMAKE_CURRENT_SOURCE_DIR}/err_opengl.hpp
+        ${CMAKE_CURRENT_SOURCE_DIR}/err_opengl.cpp
+        ${CMAKE_CURRENT_SOURCE_DIR}/font_atlas_impl.hpp
+        ${CMAKE_CURRENT_SOURCE_DIR}/font_atlas_impl.cpp
+        ${CMAKE_CURRENT_SOURCE_DIR}/font_impl.hpp
+        ${CMAKE_CURRENT_SOURCE_DIR}/font_impl.cpp
+        ${CMAKE_CURRENT_SOURCE_DIR}/gl_native_handles.hpp
+        ${CMAKE_CURRENT_SOURCE_DIR}/gl_native_handles.cpp
+        ${CMAKE_CURRENT_SOURCE_DIR}/histogram_impl.hpp
+        ${CMAKE_CURRENT_SOURCE_DIR}/histogram_impl.cpp
+        ${CMAKE_CURRENT_SOURCE_DIR}/image_impl.hpp
+        ${CMAKE_CURRENT_SOURCE_DIR}/image_impl.cpp
+        ${CMAKE_CURRENT_SOURCE_DIR}/plot_impl.hpp
+        ${CMAKE_CURRENT_SOURCE_DIR}/plot_impl.cpp
+        ${CMAKE_CURRENT_SOURCE_DIR}/surface_impl.hpp
+        ${CMAKE_CURRENT_SOURCE_DIR}/surface_impl.cpp
+        ${CMAKE_CURRENT_SOURCE_DIR}/util.cpp
+        ${CMAKE_CURRENT_SOURCE_DIR}/vector_field_impl.hpp
+        ${CMAKE_CURRENT_SOURCE_DIR}/vector_field_impl.cpp
+        ${CMAKE_CURRENT_SOURCE_DIR}/window_impl.hpp
+        ${CMAKE_CURRENT_SOURCE_DIR}/window_impl.cpp
     )
 
-LIST(SORT c_api_headers)
-LIST(SORT c_api_sources)
-LIST(SORT cpp_api_headers)
-LIST(SORT cpp_api_sources)
-LIST(SORT bknd_headers)
-LIST(SORT bknd_sources)
-LIST(SORT gl_headers)
-LIST(SORT gl_sources)
-
-SOURCE_GROUP(inc FILES ${api_headers})
-
-SOURCE_GROUP("src\\api\\c\\headers" FILES ${c_api_headers})
-SOURCE_GROUP("src\\api\\c\\sources" FILES ${c_api_sources})
-
-SOURCE_GROUP("src\\api\\cpp\\headers" FILES ${cpp_api_headers})
-SOURCE_GROUP("src\\api\\cpp\\sources" FILES ${cpp_api_sources})
-
-SOURCE_GROUP("src\\backend" FILES ${bknd_headers})
-SOURCE_GROUP("src\\backend" FILES ${bknd_sources})
-
-SOURCE_GROUP("src\\backend\\opengl\\headers" FILES ${gl_headers})
-SOURCE_GROUP("src\\backend\\opengl\\sources" FILES ${gl_sources})
-
-IF(${USE_WINDOW_TOOLKIT} STREQUAL "glfw3")
-    FILE(GLOB wtk_headers
-        "glfw/*.hpp")
-    FILE(GLOB wtk_sources
-        "glfw/*.cpp")
-    SOURCE_GROUP("src\\backend\\opengl\\glfw\\headers" FILES ${wtk_headers})
-    SOURCE_GROUP("src\\backend\\opengl\\glfw\\sources" FILES ${wtk_sources})
-ELSEIF(${USE_WINDOW_TOOLKIT} STREQUAL "sdl2")
-    FILE(GLOB wtk_headers
-        "sdl/*.hpp")
-    FILE(GLOB wtk_sources
-        "sdl/*.cpp")
-    SOURCE_GROUP("src\\backend\\opengl\\sdl\\headers" FILES ${wtk_headers})
-    SOURCE_GROUP("src\\backend\\opengl\\sdl\\sources" FILES ${wtk_sources})
-ENDIF()
-
-ADD_LIBRARY(forge SHARED
-    ${api_headers}
-    ${c_api_headers}
-    ${c_api_sources}
-    ${cpp_api_headers}
-    ${cpp_api_sources}
-    ${bknd_headers}
-    ${bknd_sources}
-    ${gl_headers}
-    ${gl_sources}
-    ${wtk_headers}
-    ${wtk_sources}
+if(${USE_WINDOW_TOOLKIT} STREQUAL "glfw3")
```

Comment:
you can make it an interface library. Interface libraries do not compile to anything but define what other libraries are created. Look at the af_graphics_interface [here](https://github.com/umar456/arrayfire/blob/cmake/CMakeLists.txt#L93). I am not creating an internal library but describing what library should include and define if they are linked against it. This will not create an libaf_graphics_interface.so library.